### PR TITLE
add prolif

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hadim** [comment]
@cbouy the test are failing with 0.3.4 and mdanalysis 2.0.0. Is that excepted, and should I disable those during the conda package build?

**Comment by hadim** [comment]
@conda-forge/staged-recipes please review

**Comment by cbouy** [comment]
Hey @hadim, ProLIF uses a [custom fork](https://github.com/cbouy/mdanalysis/tree/prolif) of MDAnalysis which hasn't yet made it to their main branch and is scheduled for v2.1.0 through [this PR](https://github.com/MDAnalysis/mdanalysis/pull/3044).

For this reason, it's not yet possible to upload prolif v0.3.4 and the previous ones to PyPI as the dependency for MDAnalysis is not `mdanalysis >=2.0.0` in the *setup.cfg* file but `mdanalysis @ git+https://github.com/cbouy/mdanalysis.git@prolif#subdirectory=package`. 
I'm not sure if conda accepts repos as dependencies, so it might have to wait for the PR to be merged.
I opened an issue for this on the ProLIF repo [here](https://github.com/chemosim-lab/ProLIF/issues/6).

Also beware that the version currently on PyPI (prolif v0.2.0) is the one from [cbouy/ProLIF](https://github.com/cbouy/ProLIF) which I archived, and the current development moved to [chemosim-lab/ProLIF](https://github.com/chemosim-lab/ProLIF) so the *about* section needs an update.

**Comment by hadim** [comment]
Thanks @cbouy. The doc at https://prolif.readthedocs.io/en/latest/source/installation.html (bottom of the page) says to wait for mdanalysis 2.0.0, so I thought it was good to go.

Since we need to wait to 2.1.0, we can't merge here for the moment.

Blocked by https://github.com/MDAnalysis/mdanalysis/issues/3446

**Comment by hadim** [comment]
@conda-forge/staged-recipes dont review and dont merge here please.

**Comment by hadim** [comment]
@cbouy do you want to review here before I ping a conda forge reviewer? Also do you want to be added as a maintainer?

**Comment by hadim** [comment]
> Won't the mdanalysis >=2.2.0 requirement be ignored when running {{ PYTHON }} -m pip install . -vv? Since the setup.cfg still contains the mdanalysis @ git+https://github.com/cbouy/mdanalysis.git@prolif#subdirectory=package requirement.

It's in the recipe and so the users installing the package will have to respect `mdanalysis >=2.2.0` which is what we want. Unless I misunderstood something?

> so might be worth to wait.

yes I think it makes more sense. Ping me here when the release is on pypi and will update the PR.

File: recipes/prolif/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - rdkit
+  run:
+    - python >=3.6
```
Comment (cbouy) [inline recipes/prolif/meta.yaml:28]:
```suggestion
    - python >=3.8
```

Comment (hadim) [reply recipes/prolif/meta.yaml:28]: The runner will set the dep required to be able to install prolif. So it's up to you to decide what min python version prolif is compatible with. If it's compatible with 3.6/3.7, then I would simply set 3.6/3.7.

Comment (hadim) [reply recipes/prolif/meta.yaml:28]: Oh sorry I thought you were talking about the `run` section when you said "the runner".

**Comment by hadim** [comment]
@cbouy i think we are good here but feel free to review again.

@conda-forge/staged-recipes ready for review

File: recipes/prolif/meta.yaml (hunk 1 above)
Comment (cbouy) [reply recipes/prolif/meta.yaml]: Yeah sorry I meant `host` :') thanks!

**Comment by hadim** [comment]
thanks @ocefpaf

**Comment by cbouy** [comment]
thanks to both of you!

**Pull request merged**